File: algorithms/Trees/Easy/LevelorderTraversal/VSDLevelorderTraversal.py

class Node:
    def __init__(self,data):
        self.data=data 
        self.left=self.right=None 



def VSDlevelorder(root):
    queue=[root]
    while queue:
        node=queue.pop(0)
        if node.left:
            queue.append(node.left)
        if node.right:
            queue.append(node.right)
        print(node.data)




if __name__=="__main__":
    root=Node(10)
    root.left=Node(20)
    root.right=Node(30)
    root.left.left=Node(40)
    root.left.right=Node(50)
    root.right.left=Node(60)
    root.right.right=Node(70)

    VSDlevelorder(root)

File: algorithms/Trees/Easy/LevelorderTraversal/VSDLevelorderTraversal.java
import java.util.LinkedList;

import java.util.Queue;

class Node {
    int data;
    Node left=null, right=null;

    Node(int data) {
        this.data = data;
    }
}

public class Main {   

    static void levelOrder(Node root){
        Queue<Node> q=new LinkedList<>();

        q.add(root);
        while(!q.isEmpty()){
            Node node=q.poll();
            if(node.left!=null)
                q.add(node.left);
            if(node.right!=null)
                q.add(node.right);
            System.out.print(node.data+" ");
        }
        
    }

   

    public static void main(String[] args) {
        Node root=new Node(10);
        root.left=new Node(20);
        root.right=new Node(30);
        root.left.left=new Node(40);
        root.left.right=new Node(50);
        root.right.left=new Node(60);
        root.right.right=new Node(70);
        levelOrder(root);
    }
}